modules/nf-core/ashlar/meta.yml

```diff
@@ -0,0 +1,44 @@
+name: "ashlar"
+description: Alignment by Simultaneous Harmonization of Layer/Adjacency Registration
+keywords:
+  - image_processing, alignment, registration
+tools:
+  #
+  - "ashlar":
+      description: "Alignment by Simultaneous Harmonization of Layer/Adjacency Registration"
+      homepage: "https://labsyspharm.github.io/ashlar/"
+      documentation: "https://labsyspharm.github.io/ashlar/"
+      doi: "10.1101/2021.04.20.440625"
+      licence: "['MIT']"
+
+input:
+  #
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', : args: '--flip-x' ]
+  - file_in:
+      type: path
+      description: multidimentional image data in formats from most commercial microscopes
+      pattern: "*.ome.tiff"
```

Comment:
Does this need to be exhaustive? There are _many_ possible extensions here. In practice no microscopes actually generate ome-tiffs -- I just converted our exemplars to that format for the sake of using an open format.

Reply:
Hmm, it might need to be exhaustive.  I'll look into this more.

Reply:
Yes, I think it's supposed to be exhaustive, but there doesn't seem to be a check when running the tests.  They state that the tests check that the input/output information between `main.nf` and `meta.yml` is consistent, but I messed around with the input file suffixes and didn't get any errors when testing.  I think we should add something like what other projects have in their `meta.yml` files like:
```
      pattern: "*.{fa,fasta,fna,fa.gz,fasta.gz,fna.gz,gbk,gbk.gz,embl,embl.gz}"
```
I've put a start of this in the last commit and it passed tests.  Can you provide a list of all or most of the possible input file extensions?  I don't think it will stop anything from working if we miss some and add them at a later date.  I should probably test this a bit more though.  Could you provide one or two small files of different types to test with?

Reading over the Ashlar documentation I realize I haven't tested for this case:
> If PATH ends in just .tif and includes {cycle} and {channel} placeholders, a series of single-channel plain TIFF files will be written.

Can you provide a good test .tif file for this?

Reply:
PATH meaning the output path, not input (as PATH is the metavar for the -o help section). E.g. if you write `ashlar inputs... -o out_{cycle}_{channel}.tif` you will get `out_0_0.tif`, `out_0_1.tif`, etc.

Reply:
```pattern: "*.{ome.tiff,ome.tif,rcpnl,btf,nd2,tif,czi}"```
We also support `*.{xdce,nd,scan,htd}` but these are trickier because those extensions just define "index" files and all the actual image tiles come in separate individual .tif files, so the whole containing directory needs to be staged not just the index file. You can look at the existing mcmicro code to see how we do this, but I might need to explain it because it's baroque :rofl: .

Reply:
See my other comments - I don't think this feature needs to be supported in this nf-core module

Reply:
Ok, sounds good.

Reply:
updated, tested and pushed.